modules/assignment/src/test/java/com/intuit/wasabi/assignment/impl/AssignmentsImplTest.java

```diff
@@ -1670,6 +1670,46 @@ public void testGetSingleAssignmentNullAssignmentRapidExperimentCapReached() {
         verify(metadataCache, times(1)).refresh();
     }
 
+    @Test
+    public void testExperimentFetchSameNameButDeleted() {
+
+        //Create input objects
+        Application.Name appName = Application.Name.valueOf("Test");
+        Experiment.ID deletedExperimentId = Experiment.ID.newInstance(); //Experiment.ID.valueOf("");
+        Experiment.ID runningExperimentId = Experiment.ID.newInstance(); //Experiment.ID.valueOf("");
+        Experiment.Label commonExperimentLabel = Experiment.Label.valueOf("SameNamePresetTwice");
+
+        Experiment deletedExperiment = mock(Experiment.class, RETURNS_DEEP_STUBS);
+        when(deletedExperiment.getID()).thenReturn(deletedExperimentId);
+        when(deletedExperiment.getEndTime().getTime()).thenReturn(new Date().getTime() + 1000000L);
+        when(deletedExperiment.getLabel()).thenReturn(commonExperimentLabel);
+        when(deletedExperiment.getState()).thenReturn(Experiment.State.DELETED);
+        when(deletedExperiment.getIsRapidExperiment()).thenReturn(false);
+
+        Experiment runningExperiment = mock(Experiment.class, RETURNS_DEEP_STUBS);
```

Comment:
@abhishekjain88 Added a static method to create mockExperiment... Note: this method potentially could be used at many places but I am not changing any existing test cases...